share/pvss/pvss_test.go

```diff
@@ -0,0 +1,310 @@
+package pvss
+
+import (
+	"testing"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/edwards"
+	"github.com/dedis/crypto/random"
+	"github.com/dedis/crypto/share"
+)
+
+func TestPVSS(test *testing.T) {
+	suite := edwards.NewAES128SHA256Ed25519(false)
+	G := suite.Point().Base()
+	H, _ := suite.Point().Pick(nil, suite.Cipher([]byte("H")))
+	n := 10
+	t := 2*n/3 + 1
+	x := make([]abstract.Scalar, n) // trustee private keys
+	X := make([]abstract.Point, n)  // trustee public keys
+	for i := 0; i < n; i++ {
+		x[i] = suite.Scalar().Pick(random.Stream)
+		X[i] = suite.Point().Mul(nil, x[i])
+	}
+
+	// Scalar of shared secret
+	secret := suite.Scalar().Pick(random.Stream)
+
+	// (1) Share distribution (dealer)
+	encShares, pubPoly, err := EncShares(suite, H, X, secret, t)
+	if err != nil {
+		test.Fatal(err)
+	}
+
+	// (2) Share decryption (trustees)
+	polys := make([]*share.PubPoly, n)
+	for i := 0; i < n; i++ {
+		polys[i] = pubPoly // NOTE: polynomials can be different
```

Comment:
Well in the case that every server is a dealer and a trustee (as in RandHound) you of course get different polynomials.